src/herbivoresAndCarnivores.js

```diff
@@ -1,15 +1,39 @@
 'use strict';
 
 class Animal {
-  // write your code here
+  static alive = [];
+  constructor(name) {
+    this.health = 100;
+    this.name = name;
+
+    Animal.alive.push(this);
+  }
+
+  die(dead) {
+    if (dead.health <= 0) {
+      Animal.alive = Animal.alive.filter((animal) => animal.health > 0);
+    }
+  }
```

Comment:
[CODE PERFORMANCE] The `die` method should not be responsible for filtering the `Animal.alive` array. It is better to perform this operation directly in the `bite` method after decreasing the health of the animal. This will avoid unnecessary checks and improve code performance.